File: src/Vts/MonteCarlo/DataStructures/PhaseFunctionInputs/HenyeyGreensteinPhaseFunctionInput.cs
using System.Runtime.Serialization;

namespace Vts.MonteCarlo
{
    /// <summary>
    /// Henyey-Greenstein phase function input
    /// </summary>
    public class HenyeyGreensteinPhaseFunctionInput : IPhaseFunctionInput
    {
        /// <summary>
        /// Henyey-Greenstein phase function
        /// </summary>
        public HenyeyGreensteinPhaseFunctionInput()
        {
            PhaseFunctionType = "HenyeyGreenstein";
        }

        /// <summary>
        /// Phase function type
        /// </summary>
        public string PhaseFunctionType { get; set; }
    }
}

File: matlab/vts_wrapper/vts/structures/MonteCarlo/TissueInputs/HenyeyGreensteinPhaseFunctionInput.m
% HENYEYGREENSTEINPHASEFUNCTIONINPUT used in MultiLayerTissueInput.m
classdef HenyeyGreensteinPhaseFunctionInput < handle % deriving from handle allows us to keep a singleton around (reference based) - see Doug's post here: http://www.mathworks.com/matlabcentral/newsreader/view_thread/171344
  properties        
        % HenyeyGreenstein phase function type 
        PhaseFunctionType = 'HenyeyGreenstein';
  end
  
  methods (Static)
      function input = FromInputNET(inputNET)
          input.PhaseFunctionType = char(inputNET.PhaseFunctionType);
      end
      
      function inputNET = ToInputNET(input)                    
          inputNET = Vts.MonteCarlo.HenyeyGreensteinPhaseFunctionInput();
      end
  end
end